MIDAS/src/finite-state-machines/fsm.cpp

```diff
@@ -0,0 +1,367 @@
+            return state;
+    
+    }
+    return state;
+}
+
+// this is similiar to the previous function but contains less states
+FSMState FSM::tick_fsm_booster(FSMState state, std::array<HighGData, 8> hg, std::array<Barometer, 8> bar){
+
+    double current_time = tick_to_ms(xTaskGetTickCount());
+    
+    switch (state.curr_state)
+    {
+        case STATE_IDLE:
+            if(getAcceleration(hg) > idle_to_first_boost_acceleration_threshold) {
+                launch_time = current_time;
+                state.curr_state = STATE_FIRST_BOOST;
+            }
+            
+            break;
+
+        case STATE_FIRST_BOOST:
+            if(getAcceleration(hg) < idle_to_first_boost_acceleration_threshold && current_time - launch_time < idle_to_first_boost_time_threshold) {
+                 state.curr_state = STATE_IDLE;
+                 break;
+            }
+            if(getAcceleration(hg) < coast_detection_acceleration_threshold) {
+                burnout_time = current_time;
+                state.curr_state  = STATE_BURNOUT;
+            }
+
+            break;
+
+        case STATE_BURNOUT:
+            if(getAcceleration(hg) >= coast_detection_acceleration_threshold && (current_time - burnout_time) < first_boost_to_burnout_time_threshold) { 
+                state.curr_state = STATE_FIRST_BOOST;
+                break;
+            }
+
+            if(current_time - burnout_time > first_boost_to_burnout_time_threshold) { 
+                sustainer_ignition_time = current_time;
+                state.curr_state = STATE_SUSTAINER_IGNITION;
+            }
+            break;
+
+        case STATE_FIRST_SEPERATION: 
+            if (getJerk(hg) < 0) {
+                state.curr_state = STATE_COAST;
+                break;
+            }
+
+            first_seperation_time = current_time;
+
+            if(current_time - first_seperation_time > first_seperation_time_threshold) { 
+	            state.curr_state = STATE_COAST;
+            }
+
+            break;
+            
+        case STATE_COAST:
+            if (getAcceleration(hg) > coast_detection_acceleration_threshold && current_time - coast_time < second_boost_to_coast_time_threshold) {
+                state.curr_state = STATE_SECOND_BOOST;
+                break;
+            }
+
+            if (getVerticalSpeed(bar) <= 0) {
+             	apogee_time = current_time;
+             	state.curr_state = STATE_APOGEE; 
+            }
+            break;
+        
+        case STATE_APOGEE:
+            if(getVerticalSpeed(bar) > 0 && current_time - apogee_time < apogee_check_threshold) {
+                state.curr_state = STATE_COAST;
+                break;
+            }
+
+            if(current_time - apogee_time > apogee_timer_threshold) {
+                drogue_time = current_time;
+                state.curr_state = STATE_DROGUE_DEPLOY;
+            }
+            break;
+            
+        case STATE_DROGUE_DEPLOY:
+            if(getJerk(hg) < 0) { 
```

Comment:
there are also a lot of places where you check if the jerk is less than 0. Realistically, this value won't be precisely zero, and will probably be some small negative value. If you could replace every 0 with a variable when you check getJerk(hg) < 0, I think that would be better